src/vm/hints/sha256_utils.rs

```diff
@@ -0,0 +1,266 @@
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    insert_value_from_var_name(
+        "full_word",
+        if n_bytes >= &bigint!(4) {
+            BigInt::one()
+        } else {
+            BigInt::zero()
+        },
+        ids,
+        &mut vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )
+}
+
+pub fn sha256_main(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let sha256_start = get_relocatable_from_var_name(
+        "sha256_start",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let input_ptr = vm.memory.get_relocatable(&sha256_start)?;
+
+    let mut message: Vec<u8> = Vec::new();
+
+    for i in 0..SHA256_INPUT_CHUNK_SIZE_FELTS {
+        message.extend(bigint_to_u32(vm.memory.get_integer(&(input_ptr + i))?)?.to_be_bytes());
+    }
+
+    let mut iv: [u32; SHA256_STATE_SIZE_FELTS] = [
+        0x6A09E667, 0xBB67AE85, 0x3C6EF372, 0xA54FF53A, 0x510E527F, 0x9B05688C, 0x1F83D9AB,
+        0x5BE0CD19,
+    ];
+
+    let new_message = GenericArray::clone_from_slice(&message);
+    compress256(&mut iv, &[new_message]);
+
+    let mut output: Vec<BigInt> = Vec::with_capacity(SHA256_STATE_SIZE_FELTS);
+
+    for new_state in iv {
+        output.push(bigint!(new_state));
+    }
+
+    let output_base = get_ptr_from_var_name(
+        "output",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    vm.segments
+        .write_arg(&mut vm.memory, &output_base, &output, Some(&vm.prime))
+        .map_err(VirtualMachineError::MemoryError)?;
+    Ok(())
+}
+
+pub fn sha256_finalize(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let message: Vec<u8> = vec![0; 64];
+
+    let mut iv: [u32; SHA256_STATE_SIZE_FELTS] = [
+        0x6A09E667, 0xBB67AE85, 0x3C6EF372, 0xA54FF53A, 0x510E527F, 0x9B05688C, 0x1F83D9AB,
+        0x5BE0CD19,
+    ];
+
+    let iv_static: Vec<BigInt> = iv.iter().map(|n| bigint!(*n)).collect();
+
+    let new_message = GenericArray::clone_from_slice(&message);
+    compress256(&mut iv, &[new_message]);
+
+    let mut output: Vec<BigInt> = Vec::with_capacity(SHA256_STATE_SIZE_FELTS);
+
+    for new_state in iv {
+        output.push(bigint!(new_state));
+    }
+
+    let sha256_ptr_end = get_ptr_from_var_name(
+        "sha256_ptr_end",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let mut padding: Vec<BigInt> = Vec::new();
+
+    for _ in 0..BLOCK_SIZE - 1 {
+        padding.extend(vec![BigInt::zero(); 16]);
+        padding.extend(iv_static.clone());
```

Comment:
```suggestion
        padding.extend_from_slice(iv_static.as_slice());
```
I remember @Oppen making this suggestion :)

Reply:
Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/357/commits/873838bc9ca19bd704f4a8af1a6bc11dfd06ca04